code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1071,6 +1071,9 @@
 	set name = "Choose Icon"
 	set waitfor = 0
 
+	if (istype(src, /mob/living/silicon/robot/shell))
```

Comment:
istyping src is bad programming practice, make a var that dictates whether it can choose its icon, or make the shell subtype's choose_icon proc return immediately